src/storage/mod.rs

```diff
@@ -2182,6 +2183,112 @@ impl TestStorageBuilder<RocksEngine, DummyLockManager> {
     }
 }
 
+#[derive(Clone)]
+pub struct TxnEngine<E: Engine> {
```

Comment:
`TxnEngine` and `TxnSnapshot` are only used for tests? I thought you added a new formal wrapper at first glance...

Reply:
Ah, yes. They are only used in tests. I will add "Test" to the names.